x-pack/plugins/apm/public/components/app/service_overview/service_overview_instances_table/index.tsx

```diff
@@ -0,0 +1,275 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexItem } from '@elastic/eui';
+import { EuiInMemoryTable } from '@elastic/eui';
+import { EuiTitle } from '@elastic/eui';
+import { EuiBasicTableColumn } from '@elastic/eui';
+import { EuiFlexGroup } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ValuesType } from 'utility-types';
+import { UNIDENTIFIED_SERVICE_NODES_LABEL } from '../../../../../common/i18n';
+import { SERVICE_NODE_NAME_MISSING } from '../../../../../common/service_nodes';
+import { useApmServiceContext } from '../../../../context/apm_service/use_apm_service_context';
+import {
+  asDuration,
+  asPercent,
+  asTransactionRate,
+} from '../../../../../common/utils/formatters';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { useUrlParams } from '../../../../context/url_params_context/use_url_params';
+import {
+  APIReturnType,
+  callApmApi,
+} from '../../../../services/rest/createCallApmApi';
+import { TruncateWithTooltip } from '../../../shared/truncate_with_tooltip';
+import { TableFetchWrapper } from '../../../shared/table_fetch_wrapper';
+import { SparkPlotWithValueLabel } from '../../../shared/charts/spark_plot/spark_plot_with_value_label';
+import { px, unit } from '../../../../style/variables';
+import { ServiceOverviewTableContainer } from '../service_overview_table_container';
+import { ServiceNodeMetricOverviewLink } from '../../../shared/Links/apm/ServiceNodeMetricOverviewLink';
+import { MetricOverviewLink } from '../../../shared/Links/apm/MetricOverviewLink';
+
+type ServiceInstanceItem = ValuesType<
+  APIReturnType<'GET /api/apm/services/{serviceName}/service_overview_instances'>
+>;
+
+interface Props {
+  serviceName: string;
+}
+
+export function ServiceOverviewInstancesTable({ serviceName }: Props) {
+  const { agentName, transactionType } = useApmServiceContext();
+
+  const {
+    urlParams: { start, end },
+    uiFilters,
+  } = useUrlParams();
+
+  const columns: Array<EuiBasicTableColumn<ServiceInstanceItem>> = [
+    {
+      field: 'name',
+      name: i18n.translate(
+        'xpack.apm.serviceOverview.instancesTableColumnNodeName',
+        {
+          defaultMessage: 'Node name',
+        }
+      ),
+      render: (_, item) => {
+        const { serviceNodeName } = item;
+        const isMissing = serviceNodeName === SERVICE_NODE_NAME_MISSING;
+        const text = isMissing
+          ? UNIDENTIFIED_SERVICE_NODES_LABEL
+          : serviceNodeName;
+
+        const link =
+          agentName === 'java' ? (
+            <ServiceNodeMetricOverviewLink
+              serviceName={serviceName}
+              serviceNodeName={item.serviceNodeName}
+            >
+              {text}
+            </ServiceNodeMetricOverviewLink>
+          ) : (
+            <MetricOverviewLink
+              serviceName={serviceName}
+              mergeQuery={(query) => ({
```

Comment:
Instead of `mergeQuery` can we just update `MetricOverviewLink` to handle the merge:

```ts
    <APMLink
      path={`/services/${serviceName}/metrics`}
      query={{
        ...query,
        ...pickKeys(urlParams as APMQueryParams, ...persistedFilters)
      }}
      {...rest}
    />
```

Reply:
I've added `mergeQuery` to `APMLink` so it's reusable. I've also made it a function to leave ordering etc up to the consumer. One might want to set a default value, or override a parameter, or simply remove one.